### ISSUE TITLE
Support explicit noindent with global `p-indent` setting

### ISSUE THREAD

**Issue by claell** [michal-h21/tex4ebook#129]
When setting `p-indent` for `tex4ebook`, it behaves similarly like pdflatex.

However, right now, it is not supported to explicitly set noindent, as described in https://tex.stackexchange.com/questions/38829/remove-indentation-for-a-single-paragraph.

Particularly, the following ways:

```latex
\noindent
Text
```

```latex
{\setlength{\parindent}{0cm}
Text
}
```

Support for this is important in case one wants to make full use of the `p-indent` option of the package.

**Comment by michal-h21** [comment]
I think that `\noindent` works, at least with this test file:

```latex
\documentclass{article}
\usepackage{lipsum}
\begin{document}

\section{hello}
\lipsum[1-3]

\noindent fourth: \lipsum[4]
\end{document}
```

**Comment by claell** [comment]
You are right. Sorry, didn't test that simple case. Here is a MWE which causes the problem:

```latex
\documentclass{article}
\usepackage{changepage}
\usepackage{lipsum}
\begin{document}

\section{hello}
\lipsum[1-3]
\noindent
\begin{adjustwidth}{2.5em}{2.5em}
    fourth: \lipsum[4]
\end{adjustwidth}
\end{document}
```

Command used:
```
tex4ebook -f epub3 test.tex p-indent
```

PDF:
![grafik](https://github.com/michal-h21/tex4ebook/assets/6171135/de8315e0-88ec-4fd9-9989-f7be5351de70)

EPUB:
![grafik](https://github.com/michal-h21/tex4ebook/assets/6171135/8384ff8f-6988-4964-a97d-b14437fcf71f)

So it seems that following environments or other things might cause `\noindent` from failing with `tex4ebook`?

**Comment by claell** [comment]
Alright. The configuration won't work for me, as I don't want to have all `adjustwidth` environments have no indent, just some. Also, it's not only about `adjustwidth`, but can also mean different environments (also nested, which might make it more tricky).

For most cases, I think that putting `\noindent` inside the lowest level of nested environments will work.

The problem is that it might happen that there are separate paragraphs inside the lowest level environment:

Like

```latex
\documentclass{article}
\usepackage{changepage}
\usepackage{lipsum}

\NewDocumentEnvironment{verticalmargins}{mm}{\vspace{#1}}{\vspace{#2}}

\begin{document}

\section{hello}
\lipsum[1-3]
{\setlength{\parindent}{0cm}
    \begin{verticalmargins}{2em}{2em}
        \begin{adjustwidth}{2.5em}{2.5em}
            fourth: \lipsum[4]

            fifth: \lipsum[5]
        \end{adjustwidth}
    \end{verticalmargins}
}
\end{document}
```

```latex
\noindent
```
does not work for this (it would need to be inserted before each paragraph).

So
```latex
{\setlength{\parindent}{0cm}
Text
}
```
is needed to set it for the complete section.

As I said, these cases are not occurring that often, and I am not sure whether they are easy to support in `tex4ebook`.

**Comment by michal-h21** [comment]
In such special cases, I would use custom environments instead of direct specification of `\setlength{parindent}` in the document. Something like this:

```latex
\documentclass{article}
\usepackage{changepage}
\usepackage{lipsum}

\NewDocumentEnvironment{verticalmargins}{mm}{\vspace{#1}}{\vspace{#2}}
\NewDocumentEnvironment{removeindent}{}{\setlength{\parindent}{0cm}}{}

\begin{document}

\section{hello}
\lipsum[1-3]
\begin{removeindent}
    \begin{verticalmargins}{2em}{2em}
        \begin{adjustwidth}{2.5em}{2.5em}
            fourth: \lipsum[4]

            fifth: \lipsum[5]
        \end{adjustwidth}
    \end{verticalmargins}
  \end{removeindent}
\end{document}
```

And corresponding config file:

```latex
\Preamble{xhtml}
\Css{.ajdustwidth p:first-child{text-indent:0}}
\ConfigureEnv{removeindent}{\ifvmode\IgnorePar\fi\EndP\HCode{<div class="removeindent">}}{\ifvmode\IgnorePar\fi\EndP\HCode{</div>}}{}{}
\RenewDocumentEnvironment{verticalmargins}{mm}{\ifvmode\IgnorePar\fi\EndP\HCode{<div class="verticalmargins" style="margin-top:#1;margin-bottom:#2;">}}{\ifvmode\IgnorePar\fi\EndP\HCode{</div>}}
\Css{.removeindent p.indent{text-indent:0;}}
\begin{document}
\EndPreamble
```

**Comment by claell** [comment]
Ah yes, good way to then detect and handle this environment in `tex4ebook`.

Now, I have some questions:
* Why is a config file used, and not a `.4ht` file? Is this case easy enough to be handled like that?
* Why is an extra definition for `adjustwidth` needed (`\Css{.ajdustwidth p:first-child{text-indent:0}}`)? In that case, the solution would not be generalized, and require extra care for every other environment that is (potentially) nested?
* Why is an extra definition for `verticalmargins` needed? Is this just because #125 is not integrated here, so it's needed to make the MWE work, but can be left out in case #125 is applied?

**Comment by michal-h21** [comment]
- you can move the configuration to a `.4ht` file. I've used a `.cfg` file only because you didn't use a custom package in the MWE. 
- You can try something like `\Css{div p:first-child{text-indent:0}}`, but it can lead to some unindeted  results.
- I reused some stuff from other answers to make this MWE work. You can copy the new configurations to your original `.4ht` file.

**Comment by claell** [comment]
Makes sense, thanks for the details! I'll play around a bit with the MWE and see how it works.
